legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/dbExtension.pure

```diff
@@ -575,7 +576,8 @@ function <<access.private>> meta::relational::functions::sqlQueryToString::proce
         let config = $sgc.config;
         let generationState = $sgc.generationState;
         let newSgc = if($func.name->in(['and', 'or']), | $sgc, | ^$sgc(config = ^$config(callingFromFilter = [])));
-        let paramWiseSgcs = if($result.toSql.parametersWithinWhenClause->isEmpty(),|range(0,$params->size(),1)->map(i|$newSgc),|$result.toSql.parametersWithinWhenClause->map(b|^$newSgc(generationState = ^$generationState(withinWhenClause=$b))));
+        let paramWiseSgcsForWithinWhenClause = if($result.toSql.parametersWithinWhenClause->isEmpty(),|range(0,$params->size(),1)->map(i|$newSgc),|$result.toSql.parametersWithinWhenClause->map(b|^$newSgc(generationState = ^$generationState(withinWhenClause=$b))));
+        let paramWiseSgcs = if($result.toSql.isBooleanNotAllowed->isEmpty(),|$paramWiseSgcsForWithinWhenClause,| $result.toSql.isBooleanNotAllowed->zip($paramWiseSgcsForWithinWhenClause)->map(b|let sgc = $b.second; ^$sgc(config = ^$config(isBooleanNotAllowed = $b.first));));
```

Comment:
instead of `let sgc = $b.second`, use `let c = $b.second` to avoid confusion with already defined sgc var defined before. Use newConfig instead of config.